### PR TITLE
POC for preservation of unicode escapes in source code

### PR DESCRIPTION
**Pull request by cal101**
This is not production ready code but touches all the places I had to change.
It works for me, see added tests.
To build it you execute the build-it script.

Some note:
- to preserve memory on parsing I used a char[]-approach
- that could be easily changed to a String reference but that would use much more memory for source code that heavily uses unicode escaped (seen with japanese/chinese coders)
- It may be worthwhile to keep the extra data only for parsing that needs it.
- I didn't looked into better javacc integration of the patches to generated files. I guess you have that in your toolbox.
- the text element currently keeps ONLY the original text I don't know if different uses cases have different needs
- if someone uses an unicode escape on ">>" or ">>>" in types they will be converted to the non-escaped variant, easy to fix but I didn't needed it in the POC

### CODE REVIEW

**Comment by cal101** [comment]
Regarding PhantomNodeLogic changes: https://github.com/ftomassetti/javaparser/issues/6

**Comment by ftomassetti** [comment]
I would discuss it after merging #124 . I think this is an important subject so I have open an issue in the original project to discuss it (https://github.com/javaparser/javaparser/issues/801)

File: build-it [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+mvn clean
+mvn clean install
+patch -p0 < ASTParser.patch
```
Comment (matozoid) [inline build-it]: I know this is a POC, but this really needs to be part of the maven build cycle through some plugin, or it should be a PR for javacc: https://github.com/javacc/javacc

**Comment by ftomassetti** [comment]
Given this is being discussed in https://github.com/javaparser/javaparser/issues/801 I will close it here.

**Pull request closed without merging**